packages/@react-aria/virtualizer/src/ScrollView.tsx

```diff
@@ -121,38 +122,33 @@ function ScrollView(props: ScrollViewProps, ref: RefObject<HTMLDivElement>) {
     };
   }, []);
 
-  useLayoutEffect(() => {
-    // TODO: resize observer
-    // https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
-    let updateSize = () => {
-      let dom = ref.current;
-      if (!dom) {
-        return;
-      }
+  let updateSize = useCallback(() => {
+    let dom = ref.current;
+    if (!dom) {
+      return;
+    }
 
-      let w = dom.clientWidth;
-      let h = dom.clientHeight;
-      if (sizeToFit && contentSize.width > 0 && contentSize.height > 0) {
-        if (sizeToFit === 'width') {
-          w = Math.min(w, contentSize.width);
-        } else if (sizeToFit === 'height') {
-          h = Math.min(h, contentSize.height);
-        }
+    let w = dom.clientWidth;
+    let h = dom.clientHeight;
+    if (sizeToFit && contentSize.width > 0 && contentSize.height > 0) {
+      if (sizeToFit === 'width') {
+        w = Math.min(w, contentSize.width);
+      } else if (sizeToFit === 'height') {
+        h = Math.min(h, contentSize.height);
       }
+    }
 
-      if (state.width !== w || state.height !== h) {
-        state.width = w;
-        state.height = h;
-        onVisibleRectChange(new Rect(state.scrollLeft, state.scrollTop, w, h));
-      }
-    };
+    if (state.width !== w || state.height !== h) {
+      state.width = w;
+      state.height = h;
+      onVisibleRectChange(new Rect(state.scrollLeft, state.scrollTop, w, h));
+    }
+  }, [onVisibleRectChange, ref, state, sizeToFit, contentSize]);
 
+  useLayoutEffect(() => {
     updateSize();
-    window.addEventListener('resize', updateSize, false);
-    return () => {
-      window.removeEventListener('resize', updateSize, false);
-    };
-  }, [onVisibleRectChange, ref, state, sizeToFit, contentSize]);
+  }, [updateSize]);
+  useResizeObserver({ref, onResize: updateSize});
```

Comment:
Not 100% sure about this but do we need the `updateSize` here in the dep array? Won't `useResizeObserver` handle calling it if `updateSize` is updated?

Reply:
useResizeObserver only attaches the listeners, it doesn't run the function, so we also need to updateSize though anytime visibleRectChange, state, sizeToFit, contentSize, ref change
https://github.com/adobe/react-spectrum/pull/910/files#diff-082f5ada66e52e3e32545d9f58296f93L150